### PR TITLE
provide for OS X system homebrew in addition to HomeBrew.jl

### PR DESCRIPTION
**Pull request by bjarthur**
if you've installed libimagemagick yourself with your own homebrew installation, this PR makes ImageMagick.jl find and use it, instead of installing libimagemagick using HomeBrew.jl.

### CODE REVIEW

**Review by timholy — APPROVED** [review]

**Comment by bjarthur** [comment]
tests are failing because of https://github.com/JuliaLang/BinDeps.jl/issues/241.  and the fix there does not work.

**Comment by bjarthur** [comment]
tests now work on all but the nightly.

**Comment by timholy** [comment]
Since I've already approved and I see no complaints, I'm hitting merge.

File: deps/build.jl [hunk 1]
```diff
@@ -91,7 +89,25 @@ function init_deps()
     ccall((:MagickWandGenesis,libwand), Void, ())
 end
 """
-    provides( Homebrew.HB, "imagemagick", libwand, os = :Darwin, preload = initfun, onload="init_deps()")
+    provides( Homebrew.HB, "imagemagick", libwand, os = :Darwin, preload = initfun_homebrew, onload="init_deps()")
+
+    if success(`brew list imagemagick`) 
+        brew_config = readlines(`brew config`);
+        idx = findfirst(x->startswith(x,"HOMEBREW_PREFIX"), brew_config)
+        brew_config[idx][18:end-1]
```
Comment (tkelman) [inline deps/build.jl:97]: this seems fragile

**Pull request merged**